src/AbstractDifferentiation.jl

```diff
@@ -372,13 +395,18 @@ function Base.:*(d::LazyHessian, ys)
     end
 
     if d.xs isa Tuple
-        return pushforward_function(
+        res =  pushforward_function(
             secondlowest(d.backend),
-            (xs...,) -> gradient(lowest(d.backend), d.f, xs...), d.xs...,)(ys)
+            (xs...,) -> gradient(lowest(d.backend), d.f, xs...)[1], d.xs...,)(ys)  # [1] because gradient returns a tuple
     else
-        return pushforward_function(
+        res =  pushforward_function(
             secondlowest(d.backend),
-            (xs,) -> gradient(lowest(d.backend), d.f, xs),d.xs,)(ys)
+            (xs,) -> gradient(lowest(d.backend), d.f, xs)[1],d.xs,)(ys)  # gradient returns a tuple
+    end
+    if res isa Tuple
```

Comment:
I am not sure if we should be using tuples at all for Hessians

Reply:
So for Hessians, you would like to return just the `matrix` and not `(matrix,)` in all cases? I had a slight preference towards returning a tuple because then AD always returns a tuple.. but if we will never include a generalization of the Hessian, it's probably fine to return `matrix`.... should I change it?